### PR TITLE
Add llama.cpp Integration

### PR DESCRIPTION
**Pull request by awinml**
### Related Issues

Part of https://github.com/deepset-ai/haystack-core-integrations/issues/177

Adds Integration page for [llama-cpp-haystack](https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/llama_cpp).

### CODE REVIEW

**Comment by bilgeyucel** [comment]
Thank you @awinml! I left some minor comments. Apart from them, it's good to go! ✅

File: integrations/llama_cpp.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+The GGUF versions of popular LLMs can be downloaded from [HuggingFace](https://huggingface.co/models?library=gguf).
+
+Initialize an `LlamaCppGenerator` with the the path to the GGUF file and also specify the required model and text generation parameters:
+
+```python
+from haystack_integrations.components.generators.llama_cpp import LlamaCppGenerator
+
+generator = LlamaCppGenerator(
+    model_path="/content/openchat-3.5-1210.Q3_K_S.gguf",
```
Comment (ZanSara) [inline integrations/llama_cpp.md:70]: See https://github.com/deepset-ai/haystack-core-integrations/pull/243
```suggestion
    model="/content/openchat-3.5-1210.Q3_K_S.gguf", 
```

File: integrations/llama_cpp.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+See [Llama.cpp's LLM documentation](https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.Llama.__init__) for more information on the available model arguments.
+
+For example, to offload the model to GPU during initialization:
+
+```python
+from haystack_integrations.components.generators.llama_cpp import LlamaCppGenerator
+
+generator = LlamaCppGenerator(
+    model_path="/content/openchat-3.5-1210.Q3_K_S.gguf",
```
Comment (ZanSara) [inline integrations/llama_cpp.md:96]: See https://github.com/deepset-ai/haystack-core-integrations/pull/243
```suggestion
    model="/content/openchat-3.5-1210.Q3_K_S.gguf",
```

File: integrations/llama_cpp.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+See [Llama.cpp's Completion API documentation](https://llama-cpp-python.readthedocs.io/en/latest/api-reference/#llama_cpp.Llama.create_completion) for more information on the available generation arguments.
+
+For example, to set the `max_tokens` and `temperature`:
+
+```python
+from haystack_integrations.components.generators.llama_cpp import LlamaCppGenerator
+
+generator = LlamaCppGenerator(
+    model_path="/content/openchat-3.5-1210.Q3_K_S.gguf",
```
Comment (ZanSara) [inline integrations/llama_cpp.md:120]: See https://github.com/deepset-ai/haystack-core-integrations/pull/243
```suggestion
    model="/content/openchat-3.5-1210.Q3_K_S.gguf",
```

File: integrations/llama_cpp.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+result = generator.run(prompt)
+```
+
+The `generation_kwargs` can also be passed to the `run` method of the generator directly:
+
+```python
+from haystack_integrations.components.generators.llama_cpp import LlamaCppGenerator
+
+generator = LlamaCppGenerator(
+    model_path="/content/openchat-3.5-1210.Q3_K_S.gguf",
```
Comment (ZanSara) [inline integrations/llama_cpp.md:136]: See https://github.com/deepset-ai/haystack-core-integrations/pull/243
```suggestion
    model="/content/openchat-3.5-1210.Q3_K_S.gguf",
```

File: integrations/llama_cpp.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+GPT4 Correct Assistant:
+"""
+
+rag_pipeline = Pipeline()
+
+text_embedder = SentenceTransformersTextEmbedder(model_name_or_path="sentence-transformers/all-MiniLM-L6-v2")
+
+# Load the LLM using LlamaCppGenerator
+model_path = "openchat-3.5-1210.Q3_K_S.gguf"
+generator = LlamaCppGenerator(model_path=model_path, n_ctx=4096, n_batch=128)
```
Comment (ZanSara) [inline integrations/llama_cpp.md:218]: See https://github.com/deepset-ai/haystack-core-integrations/pull/243
```suggestion
generator = LlamaCppGenerator(model=model_path, n_ctx=4096, n_batch=128)
```

**Review by ZanSara — COMMENTED** [review]
Hey @awinml, I recently made a small PR on this integration, so I suggested a few changes to this PR as well to keep the two in line: https://github.com/deepset-ai/haystack-core-integrations/pull/243

**Review by bilgeyucel — APPROVED** [review]

**Comment by bilgeyucel** [comment]
@ZanSara This PR is gtg. Feel free to merge it when https://github.com/deepset-ai/haystack-core-integrations/pull/243 is merge 🙌

File: integrations/llama_cpp.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+pip install llama-cpp-haystack
+```
+
+## Usage
+
+You can leverage Llama.cpp to run models by using the `LlamaCppGenerator` component.
+
+Llama.cpp requires the quantized binary of the LLM in GGUF format.
+
+The GGUF versions of popular LLMs can be downloaded from [HuggingFace](https://huggingface.co/models?library=gguf).
```
[the remark this answers is not in the document]
Comment (awinml) [reply integrations/llama_cpp.md:61]: @bilgeyucel That's a good idea! I added a section explaining how to download models from HuggingFace with an example and a code snippet for downloading the model.

**Pull request merged**